himl/config_generator.py

```diff
@@ -144,10 +144,19 @@ def dict_constructor(loader, node):
         Dumper.add_representer(OrderedDict, dict_representer)
         Loader.add_constructor(_mapping_tag, dict_constructor)
 
-        Dumper.add_representer(str, SafeRepresenter.represent_str)
+        def str_representer_pipestyle(dumper, data):
+            if '\n' in data:
```

Comment:
In order to avoid the multiple `return`s here, how about doing something like this:

```py
def str_representer_pipestyle(dumper, data):
    style = '|' if '\n' in data else None
    return dumper.represent_scalar('tag:yaml.org,2002:str', data, style=style)
```